tasks/new_e2e_tests.py

```diff
@@ -185,6 +186,27 @@ def clean(ctx, locks=True, stacks=False, output=False):
         _clean_output()
 
 
+@task
+def deps(ctx, verbose=False):
+    """
+    Setup Go dependencies
+    """
+    max_retry = 3
+    print("downloading dependencies")
+    with timed("go mod download"):
+        verbosity = ' -x' if verbose else ''
+
+        with ctx.cd("test/new-e2e"):
+            for retry in range(max_retry):
+                result = ctx.run(f"go mod download{verbosity}")
+                if result.exited is None or result.exited > 0:
+                    wait = 10 ** (retry + 1)
+                    print(f"[{(retry + 1) / max_retry}] Failed downloading, retrying in {wait} seconds")
+                    sleep(wait)
+                    continue
```

Comment:
I've already seen this somewhere 🤔

Reply:
I'll open a follow-up PR to refactor this and centralize it in a function that takes a list of path

Reply:
I'll rebase [this PR](https://github.com/DataDog/datadog-agent/pull/27465) once this one will be merged after the freeze 🙂